src/components/dialog/dialog.js

```diff
@@ -495,7 +505,18 @@ function MdDialogProvider($$interimElementProvider) {
     function onShow(scope, element, options, controller) {
       angular.element($document[0].body).addClass('md-dialog-is-showing');
 
-      wrapSimpleContent();
+      if (controller) {
+        controller.mdHtmlContent = controller.htmlContent || options.htmlContent || '';
+        controller.mdTextContent = controller.textContent || options.textContent || '';
+
+        if (controller.mdHtmlContent && !$injector.has('$sanitize')) {
+          throw Error('The ngSanitize module must be loaded in order to use htmlContent.');
```

Comment:
@jelbourn - Since dialogs are instantiated via interimElement, I believe `throw Error` never surfaces as expected. I think we should:

* make sure the exception is captured and logged in a $log 
* manually log the error also.